src/cpp/session/SessionModuleContext.cpp

```diff
@@ -95,13 +95,65 @@
 
 #include "session-config.h"
 
+#ifdef _WIN32
+# define kPathSeparator ';'
+#else
+# define kPathSeparator ':'
+#endif
+
 using namespace rstudio::core;
 using namespace boost::placeholders;
 
 namespace rstudio {
 namespace session {   
 namespace module_context {
 
+namespace {
+
+Error modifyPath(const core::FilePath& path, bool append)
+{
+   Error error;
+
+   std::string currentPath;
+   error = r::exec::RFunction("base:::Sys.getenv")
+         .addParam("PATH")
+         .call(&currentPath);
+
+   if (error)
+      return error;
+
+   std::string pathString = string_utils::utf8ToSystem(path.getAbsolutePath());
+
+#ifdef _WIN32
+   std::replace(pathString.begin(), pathString.end(), '/', '\\');
+#endif
+
+   std::string newPath = append
+         ? fmt::format("{}{}{}", currentPath, kPathSeparator, pathString)
+         : fmt::format("{}{}{}", pathString, kPathSeparator, currentPath);
+
```

Comment:
I don't think it particularly matters, but in some cases, at least on Windows I noticed I ended up with a double path separator, 
e.g.
```
D:\\Users\\jackie.gutman-win\\AppData\\Roaming\\python\\Python310\\Scripts;;D:\\Users\\jackie.gutman-win\\quarto-cli\\package\\dist\\bin;
```
(note the double semicolon) because the currentPath variable already terminated in a path separator, so it might be worth checking for that, even though I don't think it will cause any issues to have the double separator

Reply:
I'll try to take a look; I think it's worth cleaning that up